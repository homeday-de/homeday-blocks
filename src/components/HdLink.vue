<script>
export default {
  name: 'HdLink',
  props: {
    modifier: {
      type: String,
      default: 'primary',
      validator(value) {
        return [
          'primary',
          'secondary',
        ].includes(value);
      },
    },
    to: {
      type: [String, Object],
    },
  },
  computed: {
    computedClasses() {
      const baseClass = 'link';
      const classes = [
        baseClass,
        `${baseClass}--${this.modifier}`,
      ];

      return classes;
    },
  },
  render(createElement) {
    const tag = this.to ? 'router-link' : 'a';
    const data = {
      class: this.computedClasses,
    };

    if (this.to) {
      data.props = {
        to: this.to,
      };
    }

    return createElement(
      tag,
      data,
      this.$slots.default,
    );
  },
};
</script>

<style lang="scss" scoped>
</style>
