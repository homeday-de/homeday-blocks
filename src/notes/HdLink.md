You can also use it to render `<router-view>`, just provide the props `to` and it will automatically use `<router-view>` instead of default `<a>`.

```js
<template>
  <div>
    <h1>My component</h1>
    <hd-link :to="{ name: 'login' }">
      Click me
    </hd-link>
  </div>
</template>

<script>
import HdLink from 'hd-blocks/components/HdLink.vue';

export default {
  name: 'MyComponent',
  components: { HdLink },
}
</script>
```