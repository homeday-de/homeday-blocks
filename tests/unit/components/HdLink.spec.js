import { mount } from '@vue/test-utils';
import HdLink from '@/components/HdLink.vue';

describe('HdLink', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(HdLink, {
      slots: {
        default: '<span>Text</span>',
      },
    });
  });

  it('should have link class', () => {
    expect(wrapper.classes()).toContain('link');
    expect(wrapper.classes()).toContain('link--primary');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render link--secondary', () => {
    wrapper.setProps({
      modifier: 'secondary',
    });

    expect(wrapper.classes()).toContain('link--secondary');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render <router-link> on :to', () => {
    const to = 'www.homeday.de';

    wrapper.setProps({
      to,
    });

    expect(wrapper.is('router-link')).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
