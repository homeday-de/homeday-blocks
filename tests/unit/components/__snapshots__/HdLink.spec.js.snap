// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdLink should have link class 1`] = `<a class="link link--primary"><span>Text</span></a>`;

exports[`HdLink should render <router-link> on :to 1`] = `<router-link class="link link--primary"><span>Text</span></router-link>`;

exports[`HdLink should render link--secondary 1`] = `<a class="link link--secondary"><span>Text</span></a>`;
